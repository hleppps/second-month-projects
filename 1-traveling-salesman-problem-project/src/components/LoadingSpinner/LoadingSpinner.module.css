@import url(../../_variables.css);

.spinner {
  display: inline-block;
  width: 80px;
  height: 80px;
}
.spinner:after {
  content: ' ';
  display: block;
  width: 64px;
  height: 64px;
  margin-top: 10px;
  border-radius: 50%;
  border: 6px solid var(--selected-list-border-color);
  border-color: var(--selected-list-border-color) transparent var(--selected-list-border-color) transparent;
  animation: spinner 1.2s linear infinite;
}
@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
