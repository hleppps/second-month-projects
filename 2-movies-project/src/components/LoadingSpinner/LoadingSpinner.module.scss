.spinner {
  display: inline-block;
  width: 80px;
  height: 80px;
  margin: calc(50% - 40px);
  // margin-top: 40%;
}
.spinner:after {
  content: ' ';
  display: block;
  width: 64px;
  height: 64px;
  margin-top: 10px;
  border-radius: 50%;
  border: 6px solid black;
  border-color: black transparent black transparent;
  animation: spinner 1.2s linear infinite;
}
@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
