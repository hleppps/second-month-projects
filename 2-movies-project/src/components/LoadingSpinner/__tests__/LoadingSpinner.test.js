import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import LoadingSpinner from "../LoadingSpinner";

afterEach(cleanup);

it("matches a snapshot", () => {
  const { asFragment } = render(<LoadingSpinner />);
  expect(asFragment(<LoadingSpinner />)).toMatchSnapshot();
});
