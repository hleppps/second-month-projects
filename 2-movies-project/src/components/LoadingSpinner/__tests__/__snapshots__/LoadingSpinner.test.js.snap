// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner"
  />
</DocumentFragment>
`;
